### PR TITLE
Updating to 0.1.6 fdikbquery.

### PR DESCRIPTION
**Pull request by hsorby**
Also includes some work on communicating with the oSPARC platform that is not quite finished.

### CODE REVIEW

**Comment by alan-wu** [comment]
Will the testing get kicked off...?

**Comment by Tehsurfer** [comment]
Is this algood to be included? I've started working on fdikbquery@0.1.6

**Comment by hsorby** [comment]
I still have some changes pending before this gets merged.

**Comment by hsorby** [comment]
This should be ready for testing now 👍

**Comment by alan-wu** [comment]
Is fdikbquery0.1.6 the version to use for the latest changes?

**Comment by hsorby** [comment]
Yes.

**Comment by alan-wu** [comment]
Great, I will give it a try and update the UI tests if all is well.

File: map_core/app/src/main.js [hunk 1]
```diff
@@ -1,148 +1,171 @@
-var FlatmapsDialog = require('mapcoreintegratedwebapp').FlatmapsDialog;
-var FlatmapsModule = require('mapcoreintegratedwebapp').FlatmapsModule;
-var PlotsvyDialog = require('mapcoreintegratedwebapp').PlotsvyDialog;
-var PlotsvyModule = require('mapcoreintegratedwebapp').PlotsvyModule;
-var fdi_kb_query_module = require('fdikbquery').FDI_KB_Query_Module;
-var physiomeportal = require('mapcoreintegratedwebapp').physiomeportal;
+let FlatmapsDialog = require('mapcoreintegratedwebapp').FlatmapsDialog;
+let FlatmapsModule = require('mapcoreintegratedwebapp').FlatmapsModule;
+let BFCSVExporterDialog = require('mapcoreintegratedwebapp').BFCSVExporterDialog;
+let BFCSVExporterModule = require('mapcoreintegratedwebapp').BFCSVExporterModule;
```
Comment (alan-wu) [inline map_core/app/src/main.js]: This should not be changed.

File: map_core/app/src/main.js [hunk 2]
```diff
@@ -1,148 +1,171 @@
-var FlatmapsDialog = require('mapcoreintegratedwebapp').FlatmapsDialog;
-var FlatmapsModule = require('mapcoreintegratedwebapp').FlatmapsModule;
-var PlotsvyDialog = require('mapcoreintegratedwebapp').PlotsvyDialog;
-var PlotsvyModule = require('mapcoreintegratedwebapp').PlotsvyModule;
-var fdi_kb_query_module = require('fdikbquery').FDI_KB_Query_Module;
-var physiomeportal = require('mapcoreintegratedwebapp').physiomeportal;
+let FlatmapsDialog = require('mapcoreintegratedwebapp').FlatmapsDialog;
+let FlatmapsModule = require('mapcoreintegratedwebapp').FlatmapsModule;
+let BFCSVExporterDialog = require('mapcoreintegratedwebapp').BFCSVExporterDialog;
```
Comment (alan-wu) [inline map_core/app/src/main.js]: It has been renamed to PlotsvyDialog.

File: map_core/app/src/main.js [hunk 3]
```diff
@@ -167,53 +190,55 @@ main = function()  {
...
+   * {@link PJP.OrgansViewer}, {@link PJP.TissueViewer}, {@link PJP.CellPanel}
+   * and {@link PJP.ModelPanel}.
+   */
+  let initialiseMain = function () {
+    if (moduleManager) {
+      let channel = new (require('broadcast-channel')).default(channelName);
+      channel.onmessage = processMessage;
+      resizeMAPDrawingArea();
+      moduleManager.addConstructor("Flatmap", FlatmapsModule, FlatmapsDialog);
+      moduleManager.addConstructor("Data Viewer", BFCSVExporterModule, BFCSVExporterDialog);
```
Comment (alan-wu) [inline map_core/app/src/main.js]: Same as above.

File: map_core/views.py [hunk 4]
```diff
@@ -67,27 +67,18 @@ def biolucida_client_proxy(api_method=''):
...
-    # request.set_cookie("API_SESSION", value="gAAAAABdLkIw88gtEgcMAHGmlnTHafVmsAkAJRVWFSlT7j9kpDo2Jxtknxr9lQYcSnCC5UGo9IdzzyMEOPhijLvDqdyo7lEwznXmz0ZEubecPL9QohDHcUyZzMfUif8WNjZv4MXEaIKIWwfppbfSbrJBPqtewjVLSsc1PAF5UahV77qA8nyxaAip1qR1bSj-JJiBkkV4LnmP")
-    # print("==================")
-    # print(dir(request))
-    # print(request.cookies)
     return get_response_from_remote(url)
 
 
 @map_core_blueprint.route('osparc_webhook/', methods=['GET', 'POST'])
 def osparc_webhook():
+    print("response!!!!!!!")
```
Comment (alan-wu) [inline map_core/views.py:78]: Remove?

File: map_core/app/src/main.js [hunk 5]
```diff
@@ -1,148 +1,171 @@
...
+      tabManager.setTitle(data, title);
 
-			return data;
-		}
-		return undefined;
-	}
-	
-	var createDataViewer = function(organ,  annotation, url, channelNames) {
-		if (tabManager) {
-			var options = {"url":url};
```
Comment (alan-wu) [inline map_core/app/src/main.js:69]: Keep this?

File: map_core/app/src/main.js [hunk 6]
```diff
@@ -1,148 +1,171 @@
...
-			if (organ)
-				title = organ + " " + title;
+      return data;
+    }
+    return undefined;
+  };
+
+  let createDataViewer = function (organ, annotation, url, channelNames) {
+    if (tabManager) {
+      let data = tabManager.createDialog("Data Viewer");
```
Comment (alan-wu) [inline map_core/app/src/main.js]: Need the options.

**Review by alan-wu — CHANGES_REQUESTED** [review]

File: map_core/views.py (hunk 4 above)
Comment (hsorby) [reply map_core/views.py:78]: Not getting called yet still a work in progress.

**Comment by alan-wu** [comment]
Looking good and merging that in. The UI testing will catch up with this revision.

**Pull request merged**